Validate secure and unsecure access strategies (#915)
kyma-project/api-gateway: docs/user/custom-resources/apirule/04-10-apirule-custom-resource.md

>  If `service` is not defined at the **spec.service** level, all defined Access Rules must have `service` defined at the **spec.rules.service** level. Otherwise, the validation fails.

> [!WARNING]
> When you use the Ory handler, do not define the access strategies `noop`, `allow`, or `no_auth` with any other access strategy on the same **spec.rules.path**.
> When you use the Istio handler, do not define the access strategies `jwt`, `noop`, `allow`, or `no_auth` with any other access strategy on the same **spec.rules.path**.

> > Additionally, do not use secured access strategies (such as `jwt`, `oauth2_introspection`, or `oauth2_client_credentials`) with unsecured access strategies (for example, `allow`, `no_auth`, `noop`, `unauthorized`, or `anonymous`).

Reviewer: I think we should have the complete list.
```suggestion
> Additionally, do not use secured access strategies (such as `jwt`, `oauth2_introspection`, `oauth2_client_credentials`, or `cookie_session`) with unsecured access strategies (for example, `allow`, `no_auth`, `noop`, `unauthorized`, or `anonymous`).
```

> [!WARNING]

> >  Having both the Oathkeeper and Istio `jwt` access strategies defined is not supported. Access
> > strategies `noop`, `allow`, or `no_auth` cannot be used with any other access strategy on the same **spec.rules.path**.
> > Mixing secure and unsecure access strategies is not supported.

Reviewer 2: ```suggestion
> You must not use the access strategies `jwt`, `oath2_introspection`, `noop`, `allow`, or `no_auth` with any other access strategy on the same **spec.rules.path**.
> Having both the Oathkeeper and Istio `jwt` access strategies defined is not supported.
```
By 'secure access strategies' we mean `jwt` and `oath2_introspection`, right? I assume that there still are cases when the user can define more than two access strategies

> >  Having both the Oathkeeper and Istio `jwt` access strategies defined is not supported. Access
> > strategies `noop`, `allow`, or `no_auth` cannot be used with any other access strategy on the same **spec.rules.path**.
> > Mixing secure and unsecure access strategies is not supported.

Reviewer 2: ```suggestion
On the same **spec.rules.path**, you must not:
  - use the access strategies `jwt`, `oath2_introspection`, `noop`, `allow`, or `no_auth` with any other access strategy
  - define both the Oathkeeper and Istio `jwt` access strategies
  - combine secure and unsecure access strategies. The secure access strategies include `jwt` and `oath2_introspection`.
```
By 'secure access strategies' we mean `jwt` and `oath2_introspection`, right? Are there any cases when we do support defining more than one access strategy?

Reviewer: > use the access strategies `jwt`, `oath2_introspection`, `noop`, `allow`, or `no_auth` with any other access strategy.

This is not correct, you can use jwt and oauth2_introspection in the same rule.
For the Ory handler "noop", "no_auth" and "allow" can't be used with any other access strategy.
For istio handler "jwt", "noop", "no_auth" and "allow" can't be used with any other access strategy.

Reviewer 2: ```suggestion
> When you use the Ory handler, do not define the access strategies `noop`, `allow`, or `no_auth` with any other access strategy on the same **spec.rules.path**.
> When you use the Istio handler, do not define the access strategies `jwt`, `noop`, `allow`, or `no_auth` with any other access strategy on the same **spec.rules.path**..
```
Thanks, would this be fine?

Reviewer 2: I see. I edited the second suggestion and included the new validation

Author: Docs updated
